src/parsers/NumberParser.gd

```diff
@@ -50,17 +50,17 @@ static func evaluate(text: String) -> float:
 	var expr := Expression.new()
 	var err := expr.parse(text.replace(",", "."))
 	if err == OK:
-		var result: float = expr.execute()
-		if not expr.has_execute_failed():
-			return result
+		var result: String = var_to_str(expr.execute())
+		if not expr.has_execute_failed() and result.is_valid_float():
+			return str_to_var(result);
```

Comment:
Let's not have semicolons.

Reply:
Oops, lmao. every single time. I cant avoid it apparently

Reply:
Fixed